src/formMapper.js

```diff
@@ -23,18 +24,26 @@ function mapValues(elementList) {
         }
 
         if (el.matches(checkboxSelector)) {
-            if (el.checked) {
-                const currentVal = returnVal[el.name] || [];
-                returnVal[el.name] = [...currentVal, el.value];
+            if (shouldReturnAllCheckboxGroupVals) {
+                const currentVal = returnVal[el.name] || {};
+                returnVal[el.name] = {
```

Comment:
[question] Does this structure make sense for a single checkbox? For example:
`<input type="checkbox" id="agreeToTerms" name="agreeToTerms" value="true" />`

I'd expect the returned structure for that checkbox to be: 
`{
    agreeToTerms: false
}`
rather than
`{
    agreeToTerms: {
        true: false
    }
}`